src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorDataProcessingIntoKfsServiceImpl.java

```diff
@@ -85,22 +96,58 @@ private MaintenanceDocument createKfsVendorMaintenaceDocument(PaymentWorksVendor
+    private boolean kfsVendorMaintenanceDocumentValidatedForRoute(MaintenanceDocument vendorMaintenceDoc, PaymentWorksNewVendorRequestsBatchReportData reportData, PaymentWorksVendor pmwVendor) {
         boolean documentValidated = false;
         try {
             vendorMaintenceDoc.validateBusinessRules(new RouteDocumentEvent(vendorMaintenceDoc));
-            LOG.info("kfsVendorMaintenanceDocumentValidated: vendorMaintenceDoc validate.");
+            LOG.info("kfsVendorMaintenanceDocumentValidatedForRoute: vendorMaintenceDoc validate.");
+            documentValidated = true;
+        } catch (ValidationException ve) {
+            List<String> validationErrors = getPaymentWorksBatchUtilityService().convertReportDataValidationErrors(GlobalVariables.getMessageMap().getErrorMessages());
+            captureKfsProcessingErrorsForVendor(pmwVendor, reportData, validationErrors);
+            LOG.error("kfsVendorMaintenanceDocumentValidatedForRoute: eDoc validation error(s): " + validationErrors.toString());
+            LOG.error("kfsVendorMaintenanceDocumentValidatedForRoute: eDoc validation exception caught: " + ve.getMessage());       
+        } finally {
+            GlobalVariables.getMessageMap().clearErrorMessages();
+        }
+        return documentValidated;
+    }
+    
+    private boolean kfsVendorMaintenanceDocumentValidatedForSave(MaintenanceDocument vendorMaintenceDoc, PaymentWorksNewVendorRequestsBatchReportData reportData, PaymentWorksVendor pmwVendor) {
+        boolean documentValidated = false;
+        try {
+            vendorMaintenceDoc.validateBusinessRules(new SaveDocumentEvent(vendorMaintenceDoc));
+            LOG.info("kfsVendorMaintenanceDocumentValidatedForSave: vendorMaintenceDoc validate.");
             documentValidated = true;
         } catch (ValidationException ve) {
             List<String> validationErrors = getPaymentWorksBatchUtilityService().convertReportDataValidationErrors(GlobalVariables.getMessageMap().getErrorMessages());
             captureKfsProcessingErrorsForVendor(pmwVendor, reportData, validationErrors);
-            LOG.error("kfsVendorMaintenanceDocumentValidated: eDoc validation error(s): " + validationErrors.toString());
-            LOG.error("kfsVendorMaintenanceDocumentValidated: eDoc validation exception caught: " + ve.getMessage());
+            LOG.error("kfsVendorMaintenanceDocumentValidatedForSave: eDoc validation error(s): " + validationErrors.toString());
+            LOG.error("kfsVendorMaintenanceDocumentValidatedForSave: eDoc validation exception caught: " + ve.getMessage());       
         } finally {
             GlobalVariables.getMessageMap().clearErrorMessages();
         }
         return documentValidated;
     }
+    
+    private boolean saveVendorMaintenanceDocument(MaintenanceDocument vendorMaintenceDoc, PaymentWorksNewVendorRequestsBatchReportData reportData, PaymentWorksVendor pmwVendor) {
+        GlobalVariables.getMessageMap().clearErrorMessages();
```

Comment:
Is there a reason why the message map has to be cleared again here? I thought that the validation checks in this class were already clearing the map accordingly.